### PR TITLE
change MIR dump filenames from `rustc.nodeN...` to `rustc.<DefPath>`

### PR DESCRIPTION
**Pull request by mikhail-m1**
r? @nikomatsakis

### CODE REVIEW

File: src/librustc/hir/map/definitions.rs [hunk 1]
```diff
@@ -314,6 +314,26 @@ impl DefPath {
...
+    /// a TyCtxt available.
+    pub fn to_filename_friendly_no_crate(&self) -> String {
+        let mut s = String::with_capacity(self.data.len() * 16);
+
+        for component in &self.data {
+            if component.disambiguator == 0 {
+                write!(s, ".{}", component.data.as_interned_str()).unwrap();
+            } else {
+                write!(s,
+                       ".{}[{}]",
```
Comment (nikomatsakis) [inline src/librustc/hir/map/definitions.rs]: Nit: maybe use `-`? I sort of like that the `.` in the current filename allows us to "group" (e.g., `rustc.foo-bar.before.mir` can be split on `.` to get out the various bits of information about what this is a dump of)

**Comment by sinkuu** [comment]
@mikhail-m1 You may need a rebase since a new mir test ([`src/test/mir-opt/nll/named-lifetimes-basic.rs`](https://github.com/rust-lang/rust/commit/53e4bd44dedc6bbff84a5bfe564867362c320f49#diff-58c79a02a97d99d441f7241ce353a01f)) has been added in master.

**Pull request merged**